pkg/rust/cargo/naive_pkg_parser.go

```diff
@@ -0,0 +1,79 @@
+package cargo
+
+import (
+	"bufio"
+	"fmt"
+	"io"
+	"regexp"
+	"strings"
+
+	"github.com/aquasecurity/go-dep-parser/pkg/utils"
+)
+
+type pkgPosition struct {
+	start int
+	end   int
+}
+type minPkg struct {
+	name     string
+	version  string
+	position pkgPosition
+}
+
+func (pkg *minPkg) setEndPositionIfEmpty(n int) {
+	if pkg.position.end == 0 {
+		pkg.position.end = n
+	}
+}
+
+type naivePkgParser struct {
+	r io.Reader
+}
+
+var pkgDelimiterRegexp = regexp.MustCompile(`^\s*\[`)
+var pkgNameRegexp = regexp.MustCompile(`^\s*name\s=`)
+var pkgVersionRegexp = regexp.MustCompile(`^\s*version\s=`)
+var emptyLineRegexp = regexp.MustCompile(`^\s*$`)
+
+func (parser *naivePkgParser) parse() map[string]pkgPosition {
+	var currentPkg minPkg = minPkg{}
+	var idx = make(map[string]pkgPosition, 0)
+
+	scanner := bufio.NewScanner(parser.r)
+	lineNum := 1
+	for scanner.Scan() {
+		line := scanner.Text()
+		if matched := pkgDelimiterRegexp.MatchString(line); matched {
+			if currentPkg.name != "" {
+				pkgId := utils.PackageID(currentPkg.name, currentPkg.version)
+				currentPkg.setEndPositionIfEmpty(lineNum - 1)
+				idx[pkgId] = currentPkg.position
+			}
+			currentPkg = minPkg{}
+			currentPkg.position.start = lineNum
+
+		} else if matched := pkgNameRegexp.MatchString(line); matched {
+			currentPkg.name = propertyValue(line)
```

Comment:
There are times when a simple regular expression leaks memory.
What if we remove all `regexp` and use `strings` package?
Something like this:
```suggestion
		} else if strings.HasPrefix(strings.TrimSpace(line), "name") {
			currentPkg.name = propertyValue(line)
```